src/receive_message.rs

```diff
@@ -35,6 +36,46 @@ impl ReceiveMessage {
     pub fn offset(&self) -> u64 {
         self.inner.offset
     }
+
+    /// Retrieves the timestamp of the received message.
+    ///
+    /// The timestamp represents the time of the message within its topic.
+    pub fn timestamp(&self) -> u64 {
+        self.inner.timestamp
+    }
+
+    /// Retrieves the id of the received message.
+    ///
+    /// The id represents unique identifier of the message within its topic.
+    pub fn id(&self) -> u128 {
+        self.inner.id
+    }
+
+    /// Retrieves the checksum of the received message.
+    ///
+    /// The checksum represents the integrity of the message within its topic.
+    pub fn checksum(&self) -> u32 {
+        self.inner.checksum
+    }
+
+     /// Retrieves the Message's state of the received message.
+    ///
+    /// State represents the state of the response.
+    pub fn state(&self) -> String {
+        match self.inner.state {
+            RustMessageState::Available => "Available".to_string(),
```

Comment:
@lsabi maybe you could add an enum for these values instead of mapping to string?

Reply:
There's also the actual name in the `Display`. We could return a `string` but formatted the same way?

[https://github.com/iggy-rs/iggy/blob/master/sdk/src/models/messages.rs#L101](https://github.com/iggy-rs/iggy/blob/master/sdk/src/models/messages.rs#L101)